### PR TITLE
feat: start implementing parameters for each endpoint that does supports it;

### PR DESCRIPTION
**Pull request by mat1th**
Add support for query parameters for methods that do support it. 

Still need to add the tests;

### CODE REVIEW

File: Sources/MountebankSwift/Api/Endpoint/Endpoint.swift [hunk 1]
```diff
@@ -4,25 +4,44 @@ import Foundation
...
 
     /// Get a single imposter
-    static func getImposter(port: Int) -> Endpoint { Endpoint(method: .get, templatePath: "/imposters/\(port)") }
+    static func getImposter(port: Int, parameters: ImposterParameters) -> Endpoint {
+        Endpoint(method: .get, templatePath: "/imposters/\(port)", parameters: parameters)
+    }
 
     /// Create a single imposter
-    static func postImposter() -> Endpoint { Endpoint(method: .post, templatePath: "/imposters") }
+    static func postImposter(parameters: ImposterParameters) -> Endpoint {
```
Comment (teameh) [inline Sources/MountebankSwift/Api/Endpoint/Endpoint.swift:31]: Can you pass these query params when you're posting the imposters? And does that make sense?

Comment (mat1th) [reply Sources/MountebankSwift/Api/Endpoint/Endpoint.swift]: Good point, I've removed them.

File: Sources/MountebankSwift/Api/Mountebank.swift [hunk 2]
```diff
@@ -76,12 +78,18 @@ public struct Mountebank {
...
     /// `replayable` flag.
     ///
+    /// - Parameters:
+    ///   - parameters: The parameters for changing the imposter response
+    ///
     /// - Returns: A list of all ``Imposters``
     ///
     /// - Throws: `MountebankValidationError` if connection between the client and server fails in some way.
     @discardableResult
     public func getAllImposters() async throws -> Imposters {
```
Comment (teameh) [inline Sources/MountebankSwift/Api/Mountebank.swift:88]:
```suggestion
    public func getAllImposters() async throws -> [Imposter] {
``` 

Perhaps just return an array here?

File: Sources/MountebankSwift/Api/Mountebank.swift [hunk 3]
```diff
@@ -111,13 +119,18 @@ public struct Mountebank {
     ///
     /// - Parameters:
     ///   - imposter: ``Imposter`` that will be created
+    ///   - parameters: The parameters for changing the imposter response
     /// - Returns: The created ``Imposter`` with the port the ``Imposter`` will be available on.
     ///
     /// - Throws: `MountebankValidationError` if connection between the client and server fails in some way.
     @discardableResult
     public func putImposters(imposters: Imposters) async throws -> Imposters {
```
Comment (teameh) [inline Sources/MountebankSwift/Api/Mountebank.swift:127]: Perhaps just return array here? `[Imposter]`

File: Sources/MountebankSwift/Api/Mountebank.swift [hunk 4]
```diff
@@ -170,7 +187,7 @@ public struct Mountebank {
     /// - Throws: `MountebankValidationError` if connection between the client and server fails in some way.
     @discardableResult
```
Comment (teameh) [inline Sources/MountebankSwift/Api/Mountebank.swift:188]: In the comment block it says 
`Returns: The deleted Imposter`

Should be 

`- Returns: The deleted Imposters`

File: Sources/MountebankSwift/Models/Imposter/Imposters.swift [hunk 5]
```diff
@@ -3,24 +3,5 @@ import Foundation
...
-
-        enum CodingKeys: String, CodingKey {
-            case networkProtocol = "protocol"
-            case port
-        }
-    }
-
-    public let imposters: [ImposterRef]
+    public let imposters: [Imposter]
 }
```
Comment (teameh) [inline Sources/MountebankSwift/Models/Imposter/Imposters.swift:7]: I think we can get rid of this type and just use the array directly? Or make it private just for en/decodeing?

File: Tests/MountebankSwiftTests/Api/Endpoint/EndpointTests.swift [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            )): "\(Self.exampleBaseUrl.absoluteString)/imposters",
+            makeUrlFromEndpoint(.getAllImposters(
+                parameters: ImposterParameters(replayable: true)
+            )): "\(Self.exampleBaseUrl.absoluteString)/imposters?replayable=true",
+            makeUrlFromEndpoint(.getAllImposters(
+                parameters: ImposterParameters(removeProxies: false)
+            )): "\(Self.exampleBaseUrl.absoluteString)/imposters?removeProxies=false",
+        ].forEach { endpoint, expectedResult in
+            XCTAssertEqual(endpoint, URL(string: expectedResult))
+        }
```
Comment (teameh) [inline Tests/MountebankSwiftTests/Api/Endpoint/EndpointTests.swift:22]: nit: test both params together resulting in 

`/imposters?replayable=true&removeProxies=false`

File: Tests/MountebankSwiftTests/Api/Imposter+update.swift [hunk 7]
```diff
@@ -0,0 +1,83 @@
+import Foundation
+@testable import MountebankSwift
+
+extension Imposter {
+
+    func update(port: Int) -> Imposter {
```
Comment (teameh) [inline Tests/MountebankSwiftTests/Api/Imposter+update.swift:6]:
```suggestion
    func with(port: Int) -> Imposter {
```

Like this convention: 

https://developer.apple.com/documentation/musickit/musicpropertycontainer/with(_:)-10vhk/
https://developer.apple.com/documentation/uikit/uiimage/3327300-withtintcolor

**Review by teameh — APPROVED** [review]

File: Sources/MountebankSwift/Api/Mountebank.swift (hunk 2 above)
Comment (mat1th) [reply Sources/MountebankSwift/Api/Mountebank.swift]: Good point, will make the Imposters struct internal;

**Pull request merged**